### ISSUE TITLE
Likes collection in the Object seems to be missing

### ISSUE THREAD

**Issue by drequivalent** [misskey-dev/misskey#4622]
## Summary

Trying to get the object with reactions on it as a JSON representation with Curl, I see that the Likes collection is missing.
Normally, you can see the ling to the Ordered Collection page where Like activities are stored. 

## Expected Behavior

Observe this object from Pleroma:

```
$ curl -sH 'Accept: application/activity+json' https://shigusegubu.club/notice/9h22jCGtsK3Y1F2xou | jq
{
  "type": "Note",
  "to": [
    "https://pleroma.soykaf.com/users/lain",
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "tag": [
    {
      "type": "Mention",
      "name": "@lain@pleroma.soykaf.com",
      "href": "https://pleroma.soykaf.com/users/lain"
    }
  ],
  "summary": null,
  "sensitive": false,
  "repliesCount": 1,
  "published": "2019-03-22T14:32:51.080030Z",
  **"likes": {
    "type": "OrderedCollection",
    "totalItems": 2,
    "id": "https://shigusegubu.club/objects/f8e5cd08-21be-4bfd-a812-5da879f03d64/likes",
    "first": "https://shigusegubu.club/objects/f8e5cd08-21be-4bfd-a812-5da879f03d64/likes?page=1"
  },**
  "inReplyToStatusId": "9h22cQOqwqrRoHXNuC",
  "inReplyTo": "https://pleroma.soykaf.com/objects/478c2b31-8feb-4455-9235-c1f9f78c919f",
  "id": "https://shigusegubu.club/objects/f8e5cd08-21be-4bfd-a812-5da879f03d64",
  "conversation": "https://pleroma.soykaf.com/contexts/24356180-2930-41cb-9d48-bad3ef156e63",
  "context": "https://pleroma.soykaf.com/contexts/24356180-2930-41cb-9d48-bad3ef156e63",
  "content": "<span class='h-card'><a data-user='58' class='u-url mention' href='https://pleroma.soykaf.com/users/lain'>@<span>lain</span></a></span><br><a href=\"https://shigusegubu.club/media/63ac34e2e51afbc0542ee43fd27fc147809006514b38aa1774150f1cea868e06.png?name=image.png\" class='attachment'>image.png</a>",
  "cc": [
    "https://shigusegubu.club/users/hj/followers"
  ],
  "attributedTo": "https://shigusegubu.club/users/hj",
  "attachment": [
    {
      "url": "https://shigusegubu.club/media/63ac34e2e51afbc0542ee43fd27fc147809006514b38aa1774150f1cea868e06.png?name=image.png",
      "type": "Document",
      "name": "image.png",
      "mediaType": "image/png"
    }
  ],
  "actor": "https://shigusegubu.club/users/hj",
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    "https://shigusegubu.club/schemas/litepub-0.1.jsonld"
  ]
}
```
## Actual Behavior

Now, see Misskey's post:

```
$ curl -sH 'Accept: application/activity+json' https://misskey.xyz/notes/5ca13177f6ef6b0039ea0d37 | jq
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    "https://w3id.org/security/v1",
    {
      "Hashtag": "as:Hashtag"
    }
  ],
  "id": "https://misskey.xyz/notes/5ca13177f6ef6b0039ea0d37",
  "type": "Note",
  "attributedTo": "https://misskey.xyz/users/5ca0ffa5f6ef6b0039e958d0",
  "summary": null,
  "content": "<p><span>神くんだ</span></p>",
  "_misskey_content": "神くんだ",
  "published": "2019-03-31T21:30:31.588Z",
  "to": [
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "cc": [
    "https://misskey.xyz/users/5ca0ffa5f6ef6b0039e958d0/followers"
  ],
  "inReplyTo": null,
  "attachment": [],
  "sensitive": false,
  "tag": []
}
```

No Likes collection in sight, so I can't see the list of reactions to the object. That means, other Misskey instances (and all other Fediverse social engines) won't see the reactions either.

## Additional reading

https://www.w3.org/TR/activitypub/#likes

**Comment by mei23** [comment]
because **MAY**

**Comment by mei23** [comment]
Pleroma's likes collection seems to have been removed

https://git.pleroma.social/pleroma/pleroma/blob/develop/CHANGELOG.md#105-2019-08-13
> Federation: Remove likes from objects.

https://git.pleroma.social/pleroma/pleroma/commit/af4cf35e2096a6d1660271f6935b6b9ce77c6757

**Comment by mei23** [comment]
Noteにされたリアクションを列挙するやつ
https://www.w3.org/TR/activitypub/#likes

API `notes/reactions` の AP版みたいなやつ

**Comment by Johann150** [comment]
It seems Pleroma does not send the `likes` collection any more, so I don't think this is necessary. I also tested it with Mastodon which also does not seem to send this.
<details>
<summary>example from above fetched again</summary>

```shellsession
$ curl -sH 'Accept: application/activity+json' https://shigusegubu.club/objects/f8e5cd08-21be-4bfd-a812-5da879f03d64 | python -m json.tool
{
    "@context": [
        "https://www.w3.org/ns/activitystreams",
        "https://shigusegubu.club/schemas/litepub-0.1.jsonld",
        {
            "@language": "und"
        }
    ],
    "actor": "https://shigusegubu.club/users/hj",
    "attachment": [
        {
            "mediaType": "image/png",
            "name": "image.png",
            "type": "Document",
            "url": "https://shigusegubu.club/media/63ac34e2e51afbc0542ee43fd27fc147809006514b38aa1774150f1cea868e06.png?name=image.png"
        }
    ],
    "attributedTo": "https://shigusegubu.club/users/hj",
    "cc": [
        "https://shigusegubu.club/users/hj/followers"
    ],
    "content": "<span class='h-card'><a data-user='58' class='u-url mention' href='https://pleroma.soykaf.com/users/lain'>@<span>lain</span></a></span><br><a href=\"https://shigusegubu.club/media/63ac34e2e51afbc0542ee43fd27fc147809006514b38aa1774150f1cea868e06.png?name=image.png\" class='attachment'>image.png</a>",
    "context": "https://pleroma.soykaf.com/contexts/24356180-2930-41cb-9d48-bad3ef156e63",
    "conversation": "https://pleroma.soykaf.com/contexts/24356180-2930-41cb-9d48-bad3ef156e63",
    "id": "https://shigusegubu.club/objects/f8e5cd08-21be-4bfd-a812-5da879f03d64",
    "inReplyTo": "https://pleroma.soykaf.com/objects/478c2b31-8feb-4455-9235-c1f9f78c919f",
    "inReplyToStatusId": "9h22cQOqwqrRoHXNuC",
    "published": "2019-03-22T14:32:51.080030Z",
    "repliesCount": 1,
    "summary": null,
    "tag": [
        {
            "href": "https://pleroma.soykaf.com/users/lain",
            "name": "@lain@pleroma.soykaf.com",
            "type": "Mention"
        }
    ],
    "to": [
        "https://pleroma.soykaf.com/users/lain",
        "https://www.w3.org/ns/activitystreams#Public"
    ],
    "type": "Note"
}
```
</details>

<details>
<summary>example post from Gargron</summary>

```shellsession
$ curl -sH 'Accept: application/activity+json' https://mastodon.social/@Gargron/108316939822510040 | python -m json.tool
{
    "@context": [
        "https://www.w3.org/ns/activitystreams",
        {
            "ostatus": "http://ostatus.org#",
            "atomUri": "ostatus:atomUri",
            "inReplyToAtomUri": "ostatus:inReplyToAtomUri",
            "conversation": "ostatus:conversation",
            "sensitive": "as:sensitive",
            "toot": "http://joinmastodon.org/ns#",
            "votersCount": "toot:votersCount"
        }
    ],
    "id": "https://mastodon.social/users/Gargron/statuses/108316939822510040",
    "type": "Note",
    "summary": null,
    "inReplyTo": null,
    "published": "2022-05-17T11:02:16Z",
    "url": "https://mastodon.social/@Gargron/108316939822510040",
    "attributedTo": "https://mastodon.social/users/Gargron",
    "to": [
        "https://www.w3.org/ns/activitystreams#Public"
    ],
    "cc": [
        "https://mastodon.social/users/Gargron/followers"
    ],
    "sensitive": false,
    "atomUri": "https://mastodon.social/users/Gargron/statuses/108316939822510040",
    "inReplyToAtomUri": null,
    "conversation": "tag:mastodon.social,2022-05-17:objectId=293804820:objectType=Conversation",
    "content": "<p>Just realized &quot;Get started&quot; and &quot;Sign in&quot; on the first screen of the iOS app got translated into &quot;First steps&quot; and &quot;Sign up&quot; in German which explains why I&#39;ve been getting complaints that there&#39;s no way to sign in to an existing account \u00f0\u0178\u00a4\u00a6\u00e2\u20ac\udc8d\u00e2\u2122\u201a\u00ef\u00b8\udc8f</p>",
    "contentMap": {
        "en": "<p>Just realized &quot;Get started&quot; and &quot;Sign in&quot; on the first screen of the iOS app got translated into &quot;First steps&quot; and &quot;Sign up&quot; in German which explains why I&#39;ve been getting complaints that there&#39;s no way to sign in to an existing account \u00f0\u0178\u00a4\u00a6\u00e2\u20ac\udc8d\u00e2\u2122\u201a\u00ef\u00b8\udc8f</p>"
    },
    "attachment": [],
    "tag": [],
    "replies": {
        "id": "https://mastodon.social/users/Gargron/statuses/108316939822510040/replies",
        "type": "Collection",
        "first": {
            "type": "CollectionPage",
            "next": "https://mastodon.social/users/Gargron/statuses/108316939822510040/replies?only_other_accounts=true&page=true",
            "partOf": "https://mastodon.social/users/Gargron/statuses/108316939822510040/replies",
            "items": []
        }
    }
}
```
</details>

**Comment by YamiOdymel** [comment]
Mastodon **does** output the `likes` object in the Note object again 🫠

```diff
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {
      "ostatus": "http://ostatus.org#",
      "atomUri": "ostatus:atomUri",
      "inReplyToAtomUri": "ostatus:inReplyToAtomUri",
      "conversation": "ostatus:conversation",
      "sensitive": "as:sensitive",
      "toot": "http://joinmastodon.org/ns#",
      "votersCount": "toot:votersCount",
      "quote": {
        "@id": "https://w3id.org/fep/044f#quote",
        "@type": "@id"
      },
      "quoteUri": "http://fedibird.com/ns#quoteUri",
      "_misskey_quote": "https://misskey-hub.net/ns#_misskey_quote",
      "quoteAuthorization": {
        "@id": "https://w3id.org/fep/044f#quoteAuthorization",
        "@type": "@id"
      },
      "gts": "https://gotosocial.org/ns#",
      "interactionPolicy": {
        "@id": "gts:interactionPolicy",
        "@type": "@id"
      },
      "canFeature": {
        "@id": "https://w3id.org/fep/7aa9#canFeature",
        "@type": "@id"
      },
      "canQuote": {
        "@id": "gts:canQuote",
        "@type": "@id"
      },
      "automaticApproval": {
        "@id": "gts:automaticApproval",
        "@type": "@id"
      },
      "manualApproval": {
        "@id": "gts:manualApproval",
        "@type": "@id"
      }
    }
  ],
  "id": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266",
  "type": "Note",
  "summary": null,
  "inReplyTo": null,
  "published": "2026-09-07T19:34:59Z",
  "url": "https://mastodon.social/@inklingsdaily/117231432495835266",
  "attributedTo": "https://mastodon.social/users/inklingsdaily",
  "to": [
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "cc": [
    "https://mastodon.social/users/inklingsdaily/followers"
  ],
  "sensitive": false,
  "atomUri": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266",
  "inReplyToAtomUri": null,
  "conversation": "https://mastodon.social/contexts/106910914285283349-117231432495835266",
  "context": "https://mastodon.social/contexts/106910914285283349-117231432495835266",
  "content": "<p>我这辈子走路一共捡到过两块钱人民币，买彩票奖券从没中过奖，饮料都没有再来一瓶。但是我的 ex 们都是好人，我做过的工作都超有趣，我总是能交到聪明善良温柔的朋友，我长得还美，可以了。</p>",
  "contentMap": {
    "zh": "<p>我这辈子走路一共捡到过两块钱人民币，买彩票奖券从没中过奖，饮料都没有再来一瓶。但是我的 ex 们都是好人，我做过的工作都超有趣，我总是能交到聪明善良温柔的朋友，我长得还美，可以了。</p>"
  },
  "interactionPolicy": {
    "canQuote": {
      "automaticApproval": [
        "https://www.w3.org/ns/activitystreams#Public"
      ]
    }
  },
  "attachment": [],
  "tag": [],
  "replies": {
    "id": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266/replies",
    "type": "Collection",
    "first": {
      "type": "CollectionPage",
      "next": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266/replies?only_other_accounts=true&page=true",
      "partOf": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266/replies",
      "items": []
    }
  },
+  "likes": {
+    "id": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266/likes",
+    "type": "Collection",
+    "totalItems": 6
+  },
  "shares": {
    "id": "https://mastodon.social/users/inklingsdaily/statuses/117231432495835266/shares",
    "type": "Collection",
    "totalItems": 0
  }
}
```
